corehq/apps/userreports/sql/adapter.py

```diff
@@ -260,6 +260,84 @@ def doc_exists(self, doc):
             return session.query(query.exists()).scalar()
 
 
+class MultiDBSqlAdapter(object):
+
+    mirror_adapter_cls = IndicatorSqlAdapter
+
+    def __init__(self, config, override_table_name=None):
+        config.validate_db_config()
+        self.config = config
+        self.main_adapter = self.mirror_adapter_cls(config, override_table_name)
+        self.all_adapters = [self.main_adapter]
+        engine_ids = self.config.mirrored_engine_ids
+        for engine_id in engine_ids:
+            self.all_adapters.append(self.mirror_adapter_cls(config, override_table_name, engine_id))
+
+    def __getattr__(self, attr):
+        return getattr(self.main_adapter, attr)
+
+    @property
+    def table_id(self):
+        return self.config.table_id
```

Comment:
I still think many of these aren't necessary since they would fall through to the main adapter via `__getattr__`. Ideally this should only implement methods which will function differently from the main adapter.